### ISSUE TITLE
Dynamic Remarketing Tag Product ID in Merchants Feed not working

### ISSUE THREAD

**Issue by jgeerdink** [fooman/googleanalyticsplus#67]
Hi,

Our Dynamic Remarketing Tag does not work with the Product ID.
I set the correct attribute, to refer to the product ID, but when I use the Tag Assistent by Google, it displays an error.
So the Product ID is not set correctly in the code.

**Comment by fooman** [comment]
What's the error message? Also when you view the source code of your store pages what id does it show vs what it should be?

**Comment by palo** [comment]
Hi, 
Same issue here. No product ID tag.

    var google_tag_params = {
                ecomm_pagetype: 'product',
                ecomm_pvalue: 339.95,
                ecomm_category: 'Tendas',
                ecomm_totalvalue: 339.95            };

**Comment by jorgb90** [comment]
Same problem over here..

**Comment by fooman** [comment]
Can everyone please check with the latest version to confirm this fixes the issue.

**Comment by palo** [comment]
Checked the latest version, still no "ecomm_prodid" here.

**Comment by sverdier** [comment]
PHP Fatal error:  Call to a member function getDataUsingMethod() on a non-object in /srv/arrodif/arrodif/www/app/code/community/Fooman/GoogleAnalyticsPlus/Block/Remarketing.php on line 178

**Comment by the2003s** [comment]
I got the same issue on a Magento 1.7.0.2 (missing ecomm_prodid)

Could it be an issue with the extension version? Where do I check which version of GoogleAnalyticsPlus is installed?

**Comment by the2003s** [comment]
@fooman do you have a timeline for this one?

**Comment by marcosalvadori** [comment]
Hi, i have the same problem.
Do u think that u can solve it?

Thanks

**Comment by garethmorgans** [comment]
@fooman Could anyone confirm if this issue has been resolved?

**Comment by phrench** [comment]
My ecomm_prodid is populated.

Using release 0.16.8

**Comment by andrewlynch** [comment]
I've just tried installing 0.16.8 on Magento 1.9.2.1 and ecomm_prodid isn't display. Anyone else having this same is with 1.9.2.1?

**Comment by andersanmiguel** [comment]
Hi, I've solved on my case with:

On the Fooman_GoogleAnalyticsPlus_Block_Remarketing file, on the getArrayReturnValue function.

Before return the current value from the array I've had to reset the pointer:

```php
public function getArrayReturnValue($values, $default = '', $sort = false)
    {
        if (empty($values)) {
            return $default;
        }
        if ($sort) {
            asort($values);
        }

        array_walk($values, array($this, 'prepareValue'));
        if (sizeof($values) == 1) {
            reset($values);
            return current($values);
        } else {
            return '[' . implode(',', $values) . ']';
        }
    }
```

**Comment by 6Ways** [comment]
Thanks andersanmiguel!

That does the trick. For me, the product ID is now registered.

**Comment by fooman** [comment]
@6Ways we have had a reset here https://github.com/fooman/googleanalyticsplus/blob/v0.16.9/app/code/community/Fooman/GoogleAnalyticsPlus/Block/Remarketing.php#L273 for a while - could you please confirm if using were using 0.16.9

**Comment by 6Ways** [comment]
Thanks! The latest version really does work well. I had version 0.16.7 from Magento Connect.

**Comment by fooman** [comment]
Solved with 0.16.9
